internal/tools/metricslint/cmd/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"slices"
+	"strings"
+
+	"golang.org/x/tools/go/analysis/singlechecker"
+
+	"github.com/uber/cadence/internal/tools/metricslint"
+)
+
+func main() {
+	var analyze bool
+	flag.BoolVar(&analyze, "analyze", false, "if true, run the analyzer normally.  if false, run the analyzer internally and interpret the results as a linter")
+	flag.Parse()
+
+	if analyze {
+		singlechecker.Main(metricslint.Analyzer)
+	} else {
+		os.Exit(checkAnalyzerOutput())
+	}
+}
+
+func checkAnalyzerOutput() (exitCode int) {
+	cmd := exec.Command(os.Args[0], append([]string{"-analyze"}, os.Args[1:]...)...)
+	out, _ := cmd.CombinedOutput()
+
+	lines := strings.Split(string(out), "\n")
+	// map of metric name to set of lines using it (to deduplicate)
+	names := make(map[string]map[string]struct{}, len(lines))
+	var failures []string
+	for _, line := range lines {
+		if line == "" {
+			continue // empty lines are fine
+		}
+		words := strings.Fields(line)
+		if len(words) == 3 && words[1] == "success:" {
```

Comment:
this is uh... not a super fun way to parse subprocesses' output... I guess exit codes don't work, we have no other options?